### PR TITLE
Added Email SSO and Credentials Auth

### PR DESCRIPTION
**Pull request by aemmadi**
- Once user clicks log in, nextauth is triggered and handles the JWT workflow for auth.
- User has 2 options, 
  - Sign in with email
    - Uses `mailgun` to send a SSO link to email provided. The `env` variables to get it working are provided in `example.env`.
    - Once user clicks the link in email, they are redirected to homepage with a JWT session
  - Sign in with username and password (Credentials)
    - User types in their username and password
    - Right now the code for user lookup is hardcoded (`Username`: john, `Password`: test) since the database isn't populated with users. Will be updated in the future
    - This feature needs a database to even get triggered so I had to install `sqlite3` and have a basic configuration to work. Will be updated with the `MongoDB` adapter once that's ready
 - The UI has some changes dealing with this session
   - The login button automatically updates to log out when user has a valid session
   - The Navbar only displays the Account dropdown component when user has a valid session

@Joy2469 and @aashish2057 can use the example code I have provided to make this dynamic rendering of components depending on user auth status and go from there to protect pages like admin, print, etc and serve content accordingly.

TODO: Still need to figure out how to get nextauth to use the login/register page @Joy2469 created. It currently uses the default nextauth login forms.

TODO: Still working on LDAP. Should expect a PR soon with implementation, not sure how to test LDAP auth yet.
Update: [PR for LDAP Auth](#24)

### CODE REVIEW

**Review by Joy2469 — APPROVED** [review]
Awesome. It works!!!
Not gonna lie I like this UI so much. I which if it could just have the header and the footer. But we can literally keep this. It looks cool

File: pages/index.tsx [hunk 1]
```diff
@@ -6,12 +6,25 @@ import Head from "next/head";
...
+      {!session && (
+        <>
+          <IndexNavbar fixed />
+        </>
+      )}
+      {session && (
+        <>
+          <IndexNavbar fixed />
+        </>
+      )}
```
Comment (harshasrikara) [inline pages/index.tsx:27]: Doesn't this display the `<IndexNavbar />` regardless of whether we have a session or not?

File: pages/api/auth/[...nextauth].ts [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+      },
+    }),
+  ],
+  session: {
+    jwt: true,
+  },
+  callbacks: {
+    async jwt({ token, user }: { token: JWT; user: User }) {
+      const isSignIn = user ? true : false;
+      if (isSignIn) {
```
Comment (harshasrikara) [inline pages/api/auth/[...nextauth].ts:81]:
```suggestion
      if (user) {
```

File: pages/api/auth/[...nextauth].ts [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+        token.username = user.username;
+        token.password = user.password;
+      }
+      return token;
+    },
+    async session({ session, token }: { session: Session; token: User }) {
+      return { ...session, user: { username: token.username } };
```
Comment (harshasrikara) [inline pages/api/auth/[...nextauth].ts:88]:
```suggestion
      return { ...session, username: token.username  };
```

```
return { ...session, user: { ...user, username: token.username } };
```

File: pages/index.tsx [hunk 4]
```diff
@@ -6,12 +6,25 @@ import Head from "next/head";
 import Image from "next/image";
 import styles from "styles/Home.module.css";
 
+import { signIn, signOut, useSession } from "next-auth/client";
+
 import IndexNavbar from "components/Navbars/IndexNavbar";
 import Footer from "components/Footers/FooterSmall";
 const Home: NextPage = () => {
+  const [session, loading] = useSession();
```
Comment (harshasrikara) [inline pages/index.tsx:14]: add

```
useEffect(() => {
console.log(session);
}, [loading])

**Review by harshasrikara — APPROVED** [review]

**Pull request merged**